### PR TITLE
Add changelogs for 0.4 release

### PR DESCRIPTION
**Pull request by adamgundry**
See #375. This adds a few missing changelog entries. Anything else we need to change before the release?

### CODE REVIEW

**Review by phadej — COMMENTED** [review]
I admit I don't know if there is anything missing.

I don't spot any typos (I never do).

EDIT: I was supposed to click accept.

File: optics/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+* Print missing language extensions during TH generation of labels if there are
+  any
+* Add support for getters of rank1 polymorphic fields to optics generated with
+  the `makeFieldLabels` family of functions
+* Extend support of type-changing optics generated with the `makeFieldLabels`
+  family to type parameters that are phantom and applied to non-injective type
+  families
+* Fix TH generation of optics for poly-kinded data families (#378)
+* Fix `declareFieldLabels` when a field type refers to a type defined in the
+  same quote (#380)
```
Comment (phadej) [inline optics/CHANGELOG.md:27]: Note, these aren't magically turned into links. Should they be?

**Review by arybczak — APPROVED** [review]

File: optics/CHANGELOG.md (hunk 1 above)
Comment (adamgundry) [reply optics/CHANGELOG.md]: I was being lazy, but now I've done it the brute force way.

**Comment by adamgundry** [comment]
I had missed a few things, so these should be better. I'll release soon.

We should try to maintain changelogs/since pragmas in each PR, rather than rediscovering everything after the fact.

**Pull request merged**